Shared/Samples/Show realistic light and shadows/ShowRealisticLightAndShadowsView.swift

```diff
@@ -0,0 +1,144 @@
+
+import SwiftUI
+import ArcGIS
+
+struct ShowRealisticLightAndShadowsView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether the settings view should be presented.
+    @State private var isShowingSettings = false
+    
+    /// The date second value controlled by the slider.
+    @State private var dateSecond: Float = Model.dateSecondBeforeNoon
+    
+    /// The formatted text of the date controlled by the slider.
+    @State private var dateTimeText: String = DateFormatter.localizedString(
+        from: Date.now,
+        dateStyle: .medium,
+        timeStyle: .short
+    )
+    
+    /// The atmosphere effect of the scene view.
+    @State private var atmosphereEffect: SceneView.AtmosphereEffect = .realistic
+    
+    /// The sun lighting mode of the scene view.
+    @State private var lightingMode: SceneView.SunLighting = .lightAndShadows
+    
+    /// The sun date that gets passed into the scene view.
+    @State private var sunDate: Date = Date.now.advanced(by: TimeInterval(Model.dateSecondBeforeNoon))
+    
+    var body: some View {
+        VStack {
+            SceneView(scene: model.scene)
+                .atmosphereEffect(atmosphereEffect)
+                .sunLighting(lightingMode)
+                .sunDate(sunDate)
+            
+                Slider(value: $dateSecond, in: Model.dateSecondValueRange) {
+                } minimumValueLabel: {
+                    Text("AM")
+                } maximumValueLabel: {
+                    Text("PM")
+                }
+                .frame(maxWidth: 540)
+                .onChange(of: dateSecond, perform: sliderValueChanged(toValue:))
+                .padding()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Text(dateTimeText)
+                Button("Mode") {
+                    isShowingSettings = true
+                }
+                .confirmationDialog("Choose a lighting mode for the scene view.", isPresented: $isShowingSettings) {
+                    ForEach(SceneView.SunLighting.allCases, id: \.self) { mode in
+                        Button(mode.label) {
+                            lightingMode = mode
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Handles slider value changed event and set the scene view's sun date.
+    /// - Parameter value: The slider's value.
+    private func sliderValueChanged(toValue value: Float) {
+        // A DateComponents struct to encapsulate the minute value from the slider.
+        let dateComponents = DateComponents(second: Int(value))
+        let startOfToday = Date.now
+        let date = Calendar.current.date(byAdding: dateComponents, to: startOfToday)!
+        dateTimeText = DateFormatter.localizedString(from: date, dateStyle: .medium, timeStyle: .short)
+        sunDate = date
```

Comment:
```suggestion
        let dateComponents = DateComponents(second: Int(value))
        sunDate = Calendar.current.date(byAdding: dateComponents, to: .startOfDay)!
        dateTimeText = DateFormatter.localizedString(from: sunDate, dateStyle: .medium, timeStyle: .short)
```
Up to you, I tend to keep it concise if it doesn't affect readability.